core/src/test/java/io/knotx/engine/core/KnotEngineTest.java

```diff
@@ -0,0 +1,402 @@
+import io.reactivex.Single;
+import io.reactivex.SingleSource;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import io.vertx.reactivex.core.Vertx;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(VertxExtension.class)
+public class KnotEngineTest {
+
+  @BeforeEach
+  void setUp() {
+    Assertions.assertTrue(System.getenv().containsKey("vertx.logger-delegate-factory-class-name"));
+  }
+
+  @Test
+  public void execute_whenEventWithNoKnot_expectEventWithUnprocessedStatus(
+      VertxTestContext testContext, Vertx vertx) throws Throwable {
+    // given
+    KnotFlow knotFlow = null;
+
+    // when
+    verifyExecution(testContext, vertx, knotFlow, events -> {
+      // then
+      Assertions.assertEquals(1, events.size());
+      Assertions.assertEquals(Status.UNPROCESSED, events.get(0).getStatus());
+      Assertions.assertTrue(events.get(0).getLog().getJsonArray("operations").isEmpty());
+    });
+  }
+
+  @Test
+  public void execute_whenEventWithInvalidAddressInKnotFlow_expectEventWithUnprocessedStatus(
+      VertxTestContext testContext, Vertx vertx) throws Throwable {
+    // given
+    KnotFlow knotFlow = new KnotFlow("invalidAddress", Collections.emptyMap());
+
+    // when
+    verifyExecution(testContext, vertx, knotFlow, events -> {
+      // then
+      Assertions.assertEquals(1, events.size());
+      Assertions.assertEquals(Status.UNPROCESSED, events.get(0).getStatus());
+      Assertions.assertTrue(events.get(0).getLog().getJsonArray("operations").isEmpty());
+    });
+  }
+
+  @Test
+  public void execute_whenEventAndNotProcessingKnot_expectEventWithSkippedLogEntryAndUnprocessedStatus(
+      VertxTestContext testContext, Vertx vertx) throws Throwable {
+    // given
+    createNotProcessingKnot(vertx, "aAddress");
+    KnotFlow knotFlow = new KnotFlow("aAddress", Collections.emptyMap());
+
+    // when
+    verifyExecution(testContext, vertx, knotFlow, events -> {
+      // then
+      Assertions.assertEquals(1, events.size());
+      Assertions.assertEquals(Status.UNPROCESSED, events.get(0).getStatus());
+      Assertions.assertTrue(
+          verifyLogEntries(events.get(0).getLog(), Arrays.asList(
+              Operation.of("aAddress", "RECEIVED"),
+              Operation.of("aAddress", "SKIPPED")
+          )));
+    });
+  }
+
+  @Test
+  public void execute_whenEventAndProcessingKnot_expectEventWithProcessedLogEntriesAndSuccessStatus(
+      VertxTestContext testContext, Vertx vertx) throws Throwable {
+    // given
+    createSuccessKnot(vertx, "aAddress", "next");
```

Comment:
Is that important here that success knot have `"next"` transition?
Shouldn't be `null`?
How different would be behaviour when transition here is `"next"` or `null`?